plugins/module_utils/mh/base.py

```diff
@@ -29,7 +32,9 @@ def diff_mode(self):
         return self.module._diff
 
     def __getattr__(self, attr):
-        return getattr(self.module, attr)
+        if attr in self._delegated_to_module:
+            return getattr(self.module, attr)
+        return getattr(object, attr)
```

Comment:
How about this instead:
```suggestion
        raise AttributeError("ModuleHelperBase has no attribute '%s'" % (attr, ))
```
The error message `ModuleHelperBase has no attribute 'foo'` is IMO more readable than `type object 'object' has no attribute 'foo'`.